src/main.rs

```diff
@@ -107,24 +105,43 @@ fn main() {
     settings.quiet = matches.is_present("quiet");
     let vm = VirtualMachine::new(settings);
 
-    let res = import::init_importlib(&vm, true);
-    handle_exception(&vm, res);
-
-    // Figure out if a -c option was given:
-    let result = if let Some(command) = matches.value_of("c") {
-        run_command(&vm, command.to_string())
-    } else if let Some(module) = matches.value_of("m") {
-        run_module(&vm, module)
-    } else {
-        // Figure out if a script was passed:
-        match matches.value_of("script") {
-            None => run_shell(&vm),
-            Some(filename) => run_script(&vm, filename),
+    let res = (|| {
```

Comment:
This closure seems redundant. Please either remove it, or create a seperate function. Seperate function might be better, given the size of this function.